pkg/restore/restore.go

```diff
@@ -1249,13 +1249,33 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		errs.Add(namespace, err)
 		return warnings, errs
 	}
-	if isAlreadyExistsError {
-		fromCluster, err := resourceClient.Get(name, metav1.GetOptions{})
-		if err != nil {
-			ctx.log.Infof("Error retrieving cluster version of %s: %v", kube.NamespaceAndName(obj), err)
-			warnings.Add(namespace, err)
+	// check if we want to treat the error as a warning, in some cases the creation call might not get executed due to object API validations
+	// and Velero might not get the already exists error type but in reality the object already exists
+	objectExists := false
+	var fromCluster *unstructured.Unstructured
+
+	if !isAlreadyExistsError {
+		fromCluster, err = resourceClient.Get(name, metav1.GetOptions{})
+		if !apierrors.IsNotFound(err) {
```

Comment:
I think this should be `if err == nil`. If the Get errors out for any reason, we can't conclude that it exists.

Reply:
At this point, we really don't care whether it's not found or the get call failed -- either way, we're not going down the "already exists" route.